src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,119 @@
+import { useSearchParams } from 'react-router-dom';
+import { useEffect, useMemo, useState } from 'react';
+
+import { Loader } from '../components/Loader';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { PeopleTable } from '../components/PeopleTable';
+
+import { Person } from '../types';
+
+import { getPeople } from '../api';
+
+import { SexFilterType } from '../enums/SexFilterType';
+import { SortType } from '../enums/SortType';
+
+import { findPerson } from '../helpers/findPerson';
+import { getSexFilterType } from '../helpers/getSexFilterType';
+import { getSortType } from '../helpers/getSortType';
+import { getVisiblePeople } from '../helpers/getVisiblePeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+  const [hasNoDataError, setHasNoDataError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  const sexFilter = getSexFilterType(
+    searchParams.get('sex') || SexFilterType.All,
+  );
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sortType = getSortType(searchParams.get('sort') || SortType.None);
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    (async () => {
+      try {
+        setIsLoading(true);
+        const fetchedPeople = await getPeople();
+        const isEmptyResponse = !fetchedPeople.length;
+
+        if (isEmptyResponse !== hasNoDataError) {
+          setHasNoDataError(isEmptyResponse);
+        }
+
+        setPeople(
+          fetchedPeople.map((person) => {
+            const currentPerson = { ...person };
+            const { fatherName, motherName } = person;
+
+            if (motherName) {
+              currentPerson.mother = findPerson(fetchedPeople, motherName);
+            }
+
+            if (fatherName) {
+              currentPerson.father = findPerson(fetchedPeople, fatherName);
+            }
+
+            return currentPerson;
+          }),
+        );
+      } catch (err) {
+        setHasLoadingError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    })();
+  }, []);
+
+  const visiblePeople = useMemo(() => {
+    return getVisiblePeople(
+      people,
+      sexFilter,
+      query,
+      centuries.map((century) => Number(century)),
+      sortType,
+      order,
+    );
+  }, [people, sexFilter, query, centuries.length, sortType, order]);
```

Comment:
Why `centuries.length` and not `centuries`? Length can be the same while order or values were changed (not likely to happen in this demo, but avoid such logic that can be unstable)